utils/markdown/autolink.go

```diff
@@ -78,6 +78,10 @@ func parseURLAutolink(data string, position int) (Range, bool) {
 		start -= 1
 	}
 
+	if start < 0 || position >= len(data) {
```

Comment:
nit: you could do the `position >= len(data)` check in the first sanity check at the top.

Reply:
0/5 but the sanity check at the top isn't checking array bounds, I'd rather keep that together in one place